lib/features/artifacts.js

```diff
@@ -142,7 +153,9 @@ export default class Artifacts {
       var res = yield waitForEvent(putReq, 'response');
```

Comment:
Well, in the case of a regular upload, you're right, it could hang forever.  In the case of an abort, the uploadArtifact method will return and the task will complete, however there will be lingering things outside of the task waiting for that response still.